lib/di.dart

```diff
@@ -497,8 +498,30 @@ Future<void> setup({
       settingsStore: getIt.get<SettingsStore>(),
       fiatConvertationStore: getIt.get<FiatConversionStore>()));
 
+  getIt.registerFactory(
+    () => ExchangeViewModel(
+      getIt.get<AppStore>(),
+      _tradesSource,
+      getIt.get<ExchangeTemplateStore>(),
+      getIt.get<TradesStore>(),
+      getIt.get<AppStore>().settingsStore,
+      getIt.get<SharedPreferences>(),
+      getIt.get<ContactListViewModel>(),
+      getIt.get<FeesViewModel>(),
+    ),
+  );
+
+  getIt.registerSingleton(
+    SwapManager(
+      tradesStore: getIt.get<TradesStore>(),
+      settingsStore: getIt.get<SettingsStore>(),
+    ),
+  );
+
   getIt.registerFactory(() => DashboardViewModel(
       balanceViewModel: getIt.get<BalanceViewModel>(),
+      exchangeViewModel: getIt.get<ExchangeViewModel>(),
```

Comment:
this will trigger the functions and API calls of the exchange view model while it's not needed, why even use ExchangeViewModel, you already have each provider that created a swap, so just initiate this provider when you find a trade for it, and check its status